compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1163,6 +1163,426 @@ TR::Register *OMR::X86::TreeEvaluator::cstoreEvaluator(TR::Node *node, TR::CodeG
+
+   generateLabelInstruction(LABEL, node, lessThanLabel, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 2, cg);
+   
+   deps->stopAddingConditions();
+
+   generateLabelInstruction(LABEL, node, endResultLabel, deps, cg);
+   endResultLabel->setEndInternalControlFlow();
+
+   node->setRegister(resultRegister);
+
+   cg->decReferenceCount(src1AddrNode);
+   cg->decReferenceCount(src2AddrNode);
+   cg->recursivelyDecReferenceCount(lengthNode);
+
+   return resultRegister;
+   }
+
+static TR::Register *ConstLenArraycmpLenEvaluator(TR::Node *node, TR::CodeGenerator *cg, uintptrj_t arrLen)
+   {
+   TR::Register *resultRegister = cg->allocateRegister(TR_GPR);
+   TR::Node *src1AddrNode = node->getChild(0);
+   TR::Node *src2AddrNode = node->getChild(1);
+   TR::Node *lengthNode = node->getChild(2);
+   
+   int32_t byteLen = 4;
+   if (TR::Compiler->target.is64Bit())
+      byteLen = 8;
+
+   TR::Register *tempSrc1AddrRegister = cg->evaluate(src1AddrNode);;
+   TR::Register *tempSrc2AddrRegister = cg->evaluate(src2AddrNode);
+   TR::Register *tempArrayLenRegister = cg->evaluate(lengthNode);
+
+   TR::Register *src1AddrRegister = cg->allocateRegister();
+   TR::Register *src2AddrRegister = cg->allocateRegister();
+   TR::Register *arrayLenRegister = cg->allocateRegister();
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *notEqualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *backtrackLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *equalLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endResultLabel = generateLabelSymbol(cg);
+
+   //Move into registers to for use
+   generateRegRegInstruction(MOVRegReg(), node, src1AddrRegister, tempSrc1AddrRegister, cg);   
+   generateRegRegInstruction(MOVRegReg(), node, src2AddrRegister, tempSrc2AddrRegister, cg);   
+   generateRegRegInstruction(MOVRegReg(), node, arrayLenRegister, tempArrayLenRegister, cg);   
+
+   generateLabelInstruction(LABEL, node, startLabel, cg);
+   startLabel->setStartInternalControlFlow();
+
+   TR::RegisterDependencyConditions *deps = generateRegisterDependencyConditions((uint8_t) 0, 4, cg);
+   deps->addPostCondition(src1AddrRegister, TR::RealRegister::esi, cg);
+   deps->addPostCondition(src2AddrRegister, TR::RealRegister::edi, cg);
+   deps->addPostCondition(arrayLenRegister, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(resultRegister, TR::RealRegister::NoReg, cg);
+
+   //Setup
+   generateRegRegInstruction(MOVRegReg(), node, resultRegister, src1AddrRegister, cg);   
+
+   //Use CMPSQ/CMPSD to compare the first N bytes of the array that are a multiple of 8/4
+   while (arrLen >= byteLen )
+      {
+         arrLen -= byteLen ;
+         if (TR::Compiler->target.is64Bit())
+            generateInstruction(CMPSQ, node, deps, cg);
+         else 
+            generateInstruction(CMPSD, node, deps, cg);
```

Comment:
Merge into a single `generateInstruction` call as in earlier comment to save space.